varats/varats/experiments/vara/compare_traced_untraced.py

```diff
@@ -0,0 +1,275 @@
+"""Module for feature performance experiments that instrument and measure the
+execution performance of each binary that is produced by a project."""
+import os
+import typing as tp
+
+from pathlib import Path
+
+from benchbuild import Project
+from benchbuild.extensions import compiler, run, time
+from benchbuild.utils import actions
+from benchbuild.utils.cmd import time_cmd
+from plumbum import local
+
+from varats.experiment.experiment_util import (
+    exec_func_with_pe_error_handler,
+    get_default_compile_error_wrapped,
+    create_default_analysis_failure_handler,
+    ExperimentHandle,
+    get_varats_result_folder,
+    VersionExperiment,
+    get_default_compile_error_wrapped,
+)
+from varats.project.project_util import BinaryType
+from varats.provider.feature.feature_model_provider import (
+    FeatureModelProvider,
+    FeatureModelNotFound,
+)
+from varats.report.report import ReportSpecification
+from varats.report.report import FileStatusExtension as FSE
+from varats.report.report import BaseReport
+from varats.report.tef_report import TEFReport
+from varats.report.gnu_time_report import TimeReport
+
+# TODO: Refactor to use a bass class for experiments to avoid code duplication
+# where possible
+
+
+class TimeBinary(actions.Step):
+    """Executes the specified binaries and record the time it took to run"""
+
+    NAME = "TimeBinary"
+    DESCRIPTION = "Executes each binary and times it"
+
+    def __init__(self, project: Project, experiment_handle: ExperimentHandle):
+        super().__init__(obj=project, action_fn=self.time_run)  # type: ignore
+        self.__experiment_handle = experiment_handle
+
+    def time_run(self) -> actions.StepResult:
+        """Execute the specified binaries of the project, in specific
+        configurations, against one or multiple workloads."""
+        project: Project = self.obj
+
+        print(f"PWD {os.getcwd()}")
+
+        vara_result_folder = get_varats_result_folder(project)
+
+        for binary in project.binaries:  # type: ignore[attr-defined]
+            if binary.type != BinaryType.EXECUTABLE:
+                continue
+
+            result_file = self.__experiment_handle.get_file_name(
+                TimeReport.shorthand(),
+                project_name=str(project.name),
+                binary_name=binary.name,
+                project_revision=project.version_of_primary,  # type: ignore
+                project_uuid=str(project.run_uuid),
+                extension_type=FSE.SUCCESS,
+            )
+
+            with local.cwd(local.path(project.source_of_primary)):
+                print(f"Currently at {local.path(project.source_of_primary)}")
+                print(f"Bin path {binary.path}")
+                time_cmd = local["/usr/bin/time"]
+                run_cmd = time_cmd[
+                    "-o", f"{vara_result_folder}/{result_file}", "-v", binary.path
```

Comment:
```suggestion
                    "-o", f"{vara_result_folder}/{result_file}", "-v", binary["--args", "that", "are", "passed", "to", "the", "executable"]
```
Just as a hint/further information. We are currently working on making the args generic and determined by the project.
But for testing purposes, feel free to hard code here what ever you need for the projects you want to run.